pySDC/projects/PinTSimE/battery_model.py

```diff
@@ -245,23 +264,146 @@ def plot_voltages(description, problem, sweeper, use_switch_estimator, use_adapt
+        use_switch_estimator (bool): flag if the switch estimator wants to be used or not
+        recomputed (bool): flag if the values after a restart are used or before
+
+    Return:
+        data (dict): contains all information as the statistics dict
+    """
+
+    data = dict()
+
+    data['cL'] = np.array(get_sorted(stats, type='current L', recomputed=recomputed, sortby='time'))[:, 1]
+    data['vC'] = np.array(get_sorted(stats, type='voltage C', recomputed=recomputed, sortby='time'))[:, 1]
+    if use_adaptivity:
+        data['dt'] = np.array(get_sorted(stats, type='dt', recomputed=recomputed, sortby='time'))[:, 1]
+        data['e_em'] = np.array(
+            get_sorted(stats, type='error_embedded_estimate', recomputed=recomputed, sortby='time')
+        )[:, 1]
+    if use_switch_estimator:
+        data['switches'] = np.array(get_recomputed(stats, type='switch', sortby='time'))[:, 1]
+    if use_adaptivity or use_switch_estimator:
+        data['restarts'] = np.sum(np.array(get_sorted(stats, type='restart', recomputed=None, sortby='time'))[:, 1])
+    data['sum_niters'] = np.sum(np.array(get_sorted(stats, type='niter', recomputed=None, sortby='time'))[:, 1])
+
+    return data
+
+
+def get_recomputed(stats, type, sortby):
     """
+    Function that filters statistics after a recomputation. It stores all value of a type before restart. If there are multiple values
+    with same time point, it only stores the elements with unique times.
 
-    assert problem_params['alpha'] > problem_params['V_ref'], 'Please set "alpha" greater than "V_ref"'
-    assert problem_params['V_ref'] > 0, 'Please set "V_ref" greater than 0'
-    assert type(problem_params['V_ref']) == float, '"V_ref" needs to be of type float'
+    Args:
+        stats (dict): Raw statistics from a controller run
+        type (str): the type the be filtered
+        sortby (str): string to specify which key to use for sorting
 
-    assert type(problem_params['set_switch'][0]) == np.bool_, '"set_switch" has to be an bool array'
-    assert type(problem_params['t_switch']) == np.ndarray, '"t_switch" has to be an array'
-    assert problem_params['t_switch'][0] == 0, '"t_switch" is only allowed to have entry zero'
+    Returns:
+        sorted_list (list): list of filtered statistics
+    """
+
+    sorted_nested_list = []
+    times_unique = np.unique([me[0] for me in get_sorted(stats, type=type)])
+    filtered_list = [
+        filter_stats(
+            stats,
+            time=t_unique,
+            num_restarts=max([me.num_restarts for me in filter_stats(stats, type=type, time=t_unique).keys()]),
+            type=type,
+        )
+        for t_unique in times_unique
+    ]
+    for item in filtered_list:
+        sorted_nested_list.append(sort_stats(item, sortby=sortby))
+    sorted_list = [item for sub_item in sorted_nested_list for item in sub_item]
+    return sorted_list
+
+
+def proof_assertions_description(description, use_adaptivity, use_switch_estimator):
```

Comment:
maybe you could import this function from `battery_2condensators.py` and call it from here? Then you don't have to repeat the common requirements.

Reply:
Oh, yes. I see.